includes/FormFactory/ManageWikiFormFactoryBuilder.php

```diff
@@ -794,26 +795,57 @@ private static function buildDescriptorPermissions(
 				'hide-if' => [ 'OR', [ '!==', 'enable', '1' ], [ '===', 'conds', '|' ] ],
 				'default' => $aPArray[APCOND_INGROUPS] ?? [],
 				'disabled' => !$ceMW,
-				'section' => 'autopromote'
+				'section' => 'advanced'
 			]
 		];
 
 		if (
 			$ceMW &&
 			$mwPermissions->exists( $group ) &&
-			!in_array( $group, $config->get( 'ManageWikiPermissionsPermanentGroups' ) )
+			!in_array( $groupData['allGroups'], $config->get( 'ManageWikiPermissionsPermanentGroups' ) )
 		) {
+			$formDescriptor['rename-checkbox'] = [
+				'type' => 'check',
+				'label-message' => 'managewiki-permissions-rename-checkbox',
+				'disable-if' => [ '===', 'delete-checkbox', '1' ],
+				'section' => 'advanced'
+			];
+
+			$formDescriptor['rename-text'] = [
+				'type' => 'text',
+				'label-message' => 'managewiki-permissions-rename-text',
+				'help-message' => 'managewiki-permissions-rename-help',
+				'disable-if' => [ '===', 'wpdelete-checkbox', '1' ],
+				'hide-if' => [ '!==', 'rename-checkbox', '1' ],
+				'validation-callback' => [ self::class, 'validateNewGroupName' ],
+				'section' => 'advanced'
+			];
+
 			$formDescriptor['delete-checkbox'] = [
 				'type' => 'check',
-				'label-message' => 'permissions-delete-checkbox',
+				'label-message' => 'managewiki-permissions-delete-checkbox',
+				'help-message' => 'managewiki-permissions-help-delete',
 				'default' => 0,
-				'section' => 'delete'
+				'disable-if' => [ '===', 'rename-checkbox', '1' ],
+				'section' => 'advanced'
 			];
 		}
 
 		return $formDescriptor;
 	}
 
+	public static function validateNewGroupName( string $newGroup ): bool|Message {
+		if ( in_array( $newGroup, MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'managewiki' )->get( 'ManageWikiPermissionsDisallowedGroups' ) ) ) {
```

Comment:
Casing does not match:
https://github.com/miraheze/ManageWiki/blob/14d8cdf1779a482b8204bc3045a59cb98fabed62/extension.json#L296

```suggestion
		if ( in_array( $newGroup, MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'ManageWiki' )->get( 'ManageWikiPermissionsDisallowedGroups' ) ) ) {
```

Reply:
For the record this was me recently renaming the ConfigRegistry name, this was correct at the time this patch was first opened.